proof/invariant-abstract/Ipc_AI.thy

```diff
@@ -531,82 +531,8 @@ locale Ipc_AI_2 = Ipc_AI state_ext_t some_t
-  assumes make_arch_fault_msg_eq_ker_map[wp]:
-    "\<And>ft t. make_arch_fault_msg ft t \<lbrace>equal_kernel_mappings :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_asid_map [wp]:
-    "\<And>ft t. make_arch_fault_msg ft t \<lbrace>valid_asid_map :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_only_idle [wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>only_idle :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_pspace_in_kernel_window[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>pspace_in_kernel_window :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_cap_refs_in_kernel_window[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>cap_refs_in_kernel_window :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_valid_objs[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>valid_objs :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_valid_global_objs[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>valid_global_objs :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_valid_global_vspace_mappings[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>valid_global_vspace_mappings :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_valid_ioc[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>valid_ioc :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_vms[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>valid_machine_state :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_st_tcb_at'[wp]:
-    "\<And> P p ft t . make_arch_fault_msg ft t \<lbrace>st_tcb_at P p :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_cap_to[wp]:
-    "\<And> ft t p. make_arch_fault_msg ft t \<lbrace>ex_nonz_cap_to p :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_valid_irq_states[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>valid_irq_states :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_cap_refs_respects_device_region[wp]:
-    "\<And> ft t. make_arch_fault_msg ft t \<lbrace>cap_refs_respects_device_region :: 'state_ext state \<Rightarrow> bool\<rbrace>"
-  assumes make_arch_fault_msg_pred_tcb[wp]:
-    "\<And> Q P (proj :: itcb \<Rightarrow> 't) ft t p. make_arch_fault_msg ft t \<lbrace>\<lambda>s::'state_ext state. Q (pred_tcb_at proj P p s)\<rbrace>"
-  assumes make_arch_fault_msg_arch_tcb_at[wp]:
-    "\<And> Q P ft t p. make_arch_fault_msg ft t \<lbrace>\<lambda>s::'state_ext state. Q (arch_tcb_at P p s)\<rbrace>"
+  assumes make_arch_fault_msg_inv[wp]:
+    "\<And>ft t. make_arch_fault_msg ft t \<lbrace>P :: 'state_ext state \<Rightarrow> bool\<rbrace>"
```

Comment:
I'm a bit tired... I can't seem to find where you actually satisfy this assumption. Is it because a crunch is carrying [Ipc_AI_2_assms] on one or more of its properties?

Reply:
Each of the arches actually already proved it and gave it the right attribute, so I just had to change the interface. See https://github.com/seL4/l4v/blob/2cfe9da8dc5b767bafdcdd1d64a73c07f3440fb0/proof/invariant-abstract/AARCH64/ArchIpc_AI.thy#L226 for an example.